src/dosext/mfs/mfs.c

```diff
@@ -957,8 +958,11 @@ mfs_redirector(void)
 }
 
 int
-mfs_inte6(void)
+mfs_inte6(int fromDOS)
 {
+  if (fromDOS && WORD(REGS.ebx) == DOS_SUBHELPER_MFS_REDIR_INIT)
+    emufs_loaded = TRUE;
```

Comment:
Because we need to distinguish between emufs.sys being loaded and the redirector being initialised, since the latter can occur via two methods currently.

I put the thing into version check mainly because you suggested it would be a good idea.
>Also ems.sys can be loaded twice...
Would be nice if you take care of that.
I think the best would be to stuff this
into the version checking. Besides
version OK/old it can also return the
installed status.

I really don't like it myself as it seems odd that when checking the version that it an error that the emufs.sys is already loaded.
BX=0 function serves no purpose any more, it previously redirected one drive. The function 0x5 (then 0x500) did the init and is still called from both 32 bit code and Device Driver.
I'll see if I can do this all from within the Device Driver, as I like the way that ems.sys now handles this. It's a shame we don't install a unique interrupt handler like ems.sys did, then we could easily find the segment emufs.sys was loaded at and so the driver name. Possibly I could walk the Device Driver chain from the pointer in LOL?